db/crud_test.go

```diff
@@ -956,3 +956,219 @@ func BenchmarkHandleRevDelta(b *testing.B) {
+	parent := rev // Immediate ancestor or parent revision
+	assert.NoError(t, err, "Couldn't create document")
+
+	payload = `{"sku":"6213102","_attachments":{"camera.txt":{"stub":true,"revpos":1}}}`
+	doc, rev, err = db.PutExistingRevWithBody("camera", unjson(payload), []string{"3-a", "2-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+
+	// Get available attachments by immediate ancestor revision or parent revision
+	meta, found := db.getAvailableRevAttachments(doc, parent)
+	attachment := meta["camera.txt"].(map[string]interface{})
+	assert.Equal(t, "sha1-VoSNiNQGHE1HirIS5HMxj6CrlHI=", attachment["digest"])
+	assert.Equal(t, float64(20), attachment["length"])
+	assert.Equal(t, float64(1), attachment["revpos"])
+	assert.True(t, found, "Ancestor should exists")
+
+	// Get available attachments by immediate ancestor revision
+	meta, found = db.getAvailableRevAttachments(doc, ancestor)
+	attachment = meta["camera.txt"].(map[string]interface{})
+	assert.Equal(t, "sha1-VoSNiNQGHE1HirIS5HMxj6CrlHI=", attachment["digest"])
+	assert.Equal(t, float64(20), attachment["length"])
+	assert.Equal(t, float64(1), attachment["revpos"])
+	assert.True(t, found, "Ancestor should exists")
+}
+
+func TestGet1xRevAndChannels(t *testing.T) {
+	testBucket := base.GetTestBucket(t)
+	defer testBucket.Close()
+	bucket := testBucket.Bucket
+
+	context, err := NewDatabaseContext("db", bucket, false, DatabaseContextOptions{})
+	assert.NoError(t, err, "Couldn't create context for database 'db'")
+	defer context.Close()
+	db, err := CreateDatabase(context)
+	assert.NoError(t, err, "Couldn't create database 'db'")
+
+	payload := `{"sku":"6213100","_attachments":{"camera.txt":{"data":"Q2Fub24gRU9TIDVEIE1hcmsgSVY="}}}`
+	doc1, rev1, err := db.PutExistingRevWithBody("camera", unjson(payload), []string{"1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+
+	payload = `{"sku":"6213101","_attachments":{"lens.txt":{"data":"Q2Fub24gRU9TIDVEIE1hcmsgSVY="}}}`
+	doc2, rev2, err := db.PutExistingRevWithBody("camera", unjson(payload), []string{"2-a", "1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+
+	// Get the 1x revision from document with list revision enabled
+	bodyBytes, removed, err := db.get1xRevFromDoc(doc2, rev2, true)
+	assert.False(t, removed)
+	assert.NoError(t, err, "It should not throw any error")
+	assert.NotNil(t, bodyBytes, "Document body bytes should be received")
+
+	var response = Body{}
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+
+	// Get the 1x revision from document with list revision enabled
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc1, rev1, true)
+	assert.False(t, removed)
+	assert.NoError(t, err, "It should not throw any error")
+	assert.NotNil(t, bodyBytes, "Document body bytes should be received")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+
+	// Delete the revision rev2 and get the revision from doc2
```

Comment:
Comment should be 'delete the document, creating tombstone revision rev3'

Reply:
Comment has been modified and changes are available in latest revision.